api/controllers/EventController.js

```diff
@@ -2,9 +2,38 @@ var Event = require('../models/Event');
 var User = require('../models/User');
 var auth = require('../utils/auth');
 var validator = require('../utils/validator');
+var Restaurant = require('../models/Restaurant');
+var PromotionController = require('../controllers/PromotionController');
 
 var EventController = {
 
+    validateOpenHour: function(data, res, event){
+        var restaurant = data.body;
+
+        var isValid = false;
+        var startTime = new Date(event.startTime);
+        var day = startTime.getDay();
+        var time = "" + startTime.getHours() + startTime.getMinutes();
+
+        restaurant.hours.forEach(function(element) {
+            element.open.forEach(function(entry){
+                if(day == entry.day){
+                    isValid = (time > entry.start && time < entry.end);
+                }
+            })
+        });
+
+        if(isValid){
+            Event.create(event, function (err, _) {
```

Comment:
this `validateOpenHour` method does more than just validating now

Reply:
Yeah, I should move it to validator class.